v4l2/v4l2_codecbase.cpp

```diff
@@ -337,23 +337,6 @@ int32_t V4l2CodecBase::ioctl(int command, void* arg)
             ASSERT(ret != -1);
             if (port == INPUT) {
                 DEBUG("start decoding/encoding");
-#if (defined(ANDROID) || defined(__ENABLE_WAYLAND__))
-                /* there is no dedicate API for V4l2Codec flush(),
-                 * but use STREAMOFF/STREAMON to flush staging buffers.
-                 * so, some initialization work needn't twice here
-                 */
-                if (!m_started) {
-                    if (!setVaDisplay()) {
-                        ERROR("fail to set up VADisplay");
-                        ret = -1;
-                    }
-
-                    if (!createVpp()) {
-                        ERROR("fail to set up VPP");
-                        ret = -1;
-                    }
-                }
-#endif
                 boolRet = start();
```

Comment:
This conflict with your change log, Move code to start still called by decode thread.